lib/task_bunny/message.ex

```diff
@@ -1,9 +1,42 @@
 defmodule TaskBunny.Message do
   @moduledoc """
-  Provides functionalities to access a message and its meta data.
+  Functions to access messages and its meta data.
   """
 
   @doc """
+  Encode message body in JSON with job and arugment.
+  """
+  @spec encode(atom, any) :: String.t
+  def encode(job, payload) do
+    %{
+      "job" => Atom.to_string(job),
+      "payload" => payload,
+      "created_at" => DateTime.utc_now()
+    }
+    |> Poison.encode!
+  end
+
+  @doc """
+  Decode message body in JSON to map
+  """
+  @spec decode(String.t) :: map
+  def decode(message) do
+    case Poison.decode(message) do
+      {:ok, decoded} ->
+        job = String.to_atom(decoded["job"])
```

Comment:
Ahw, I figured out what the difference is:
```elixir
String.to_atom("Jobs.ExampleJob") != Jobs.ExampleJob
String.to_atom("Elixir.Jobs.ExampleJob") == Jobs.ExampleJob
```
and in the tests you are always passing the string `"Elixir.TaskBunny.MessageTest.Namespace.NameJob"`, which makes the tests pass, because they start with "Elixir." in the string.

The method I linked will accept both `"Jobs.ExampleJob"` and `"Elixir.Jobs.ExampleJob"` as valid values to decode. This one only `"Elixir.Jobs.ExampleJob"`.

Something to think about, since it is a bit "strange" for other languages that use the *job system* to have to start everything with the word: `Elixir.`.


If you do decide the keep using this method with prepending "Elixir." to the string, then I would indeed switch to `to_existing_atom`.